.github/workflows/samples-typescript-typecheck.yaml

```diff
@@ -0,0 +1,26 @@
+name: TypeScript clients type checks
+
+on:
+  pull_request:
+    paths:
+      - samples/**
```

Comment:
@joscha can we run this only if there's a change in the typescript-related samples?

https://github.com/OpenAPITools/openapi-generator/pull/20000 (a recent PR) shows the workflow failed.

can you please take a look?

Reply:
I can reduce it a bit more. In order to be super precise we'd need the full dependency graph, that is tricky as it includes any package.lock, node versions, etc. - it's not impossible but the repository is not set up for it currently, we'd want to also get the node dependency from nix for example. I can probably improve it a bit more, however I won't have access to a computer for a few more days. If the tests fail regularly with false positives, feel free to comment out the yaml and I'll fix and  reenable when I get access to a computer.